### DISCUSSION TITLE
config.txt and Linux

### DISCUSSION THREAD

**Discussion by Adephx** [jagrosh/MusicBot#1114] [Questions]
I would appreciate if you could add some extra guidelines to the wiki regarding running the JMusicBot on Linux.
For example, I have a /JMusicBot directory, so one would assume that launching the JMusicBot with the following command `nohup java  -Dnogui=true -jar /JMusicBot/JMusicBot-0.3.6.jar &` would suffice. Sadly, that's not the case, because the program creates a new config file in /home/pi/directory, because we are not cd-ed in the /JMusicBot directory. This is particularly annoying when you are tying to launch the script automatically on boot. ~~I bypassed this issue by adding a "custom" config file `nohup java -Dconfig=/JMusicBot/config.txt -Dnogui=true -jar /JMusicBot/JMusicBot-0.3.6.jar &`~~

~~I then added the scrip to /etc/rc.local~~

~~Any reason to why the program doesn't look for the config.txt in it's root directory like it does with Playlists folder?~~

Edit: It seems that the Playlists also don't get loaded from the program's root directory, my "solution":

/etc/rc.local

```(

cd /JMusicBot

nohup java -Dnogui=true -jar JMusicBot-0.3.6.jar &

)

exit 0```

**Comment by Sanduhr32** [comment]
If you want to autostart on linux properly, look into the init system used by your Linux distro of choice.

For debian based distro system-d service configurations are recommended.

Linux systems really depend on the distro & user and aren't universal.
Most Linux users know what they are doing & do not need advanced guidance, they appreciate pre-made configurations or simple guidance on "how to handle the project".
